### PR TITLE
silence compiler warnings

### PR DESCRIPTION
**Pull request by rettinghaus**
This fixes some compiler warnings in MusicXML import/export.

### CODE REVIEW

File: src/importexport/musicxml/internal/musicxml/importmxmlpass2.cpp [hunk 1]
```diff
@@ -5876,7 +5873,7 @@ void MusicXMLParserPass2::harmony(const QString& partId, Measure* measure, const
             size_t nstaves = _pass1.getPart(partId)->nstaves();
             QString strStaff = _e.readElementText();
             int staff = _pass1.getMusicXmlPart(partId).staffNumberToIndex(strStaff.toInt());
-            if (staff >= 0 && staff < nstaves) {
+            if (staff >= 0 && staff < int(nstaves)) {
```
Comment (Jojo-Schmitz) [inline src/importexport/musicxml/internal/musicxml/importmxmlpass2.cpp:5876]: I believe we'd prefer the mor verbose C++ casting `static_cast<int>(nstaves)`

File: src/importexport/musicxml/internal/musicxml/importmxmlpass2.cpp [hunk 2]
```diff
@@ -4351,7 +4351,7 @@ void MusicXMLParserPass2::clef(const QString& partId, Measure* measure, const Fr
     if (strClefno != "") {
         clefno = _pass1.getMusicXmlPart(partId).staffNumberToIndex(strClefno.toInt());
     }
-    if (clefno < 0 || clefno >= part->nstaves()) {
+    if (clefno < 0 || clefno >= int(part->nstaves())) {
```
Comment (Jojo-Schmitz) [inline src/importexport/musicxml/internal/musicxml/importmxmlpass2.cpp:4354]: see below

File: src/importexport/musicxml/internal/musicxml/importmxmlpass2.cpp [hunk 3]
```diff
@@ -2559,7 +2559,7 @@ void MusicXMLParserPass2::staffDetails(const QString& partId, Measure* measure)
     int n = 0;  // default
     if (strNumber != "") {
         n = _pass1.getMusicXmlPart(partId).staffNumberToIndex(strNumber.toInt());
-        if (n < 0 || n >= staves) {
+        if (n < 0 || n >= int(staves)) {
```
Comment (Jojo-Schmitz) [inline src/importexport/musicxml/internal/musicxml/importmxmlpass2.cpp:2562]: see below

File: src/importexport/musicxml/internal/musicxml/importmxmlpass2.cpp (hunk 1 above)
Comment (rettinghaus) [reply src/importexport/musicxml/internal/musicxml/importmxmlpass2.cpp:5876]: The most cases (at least in this file) are using the simple cast.

**Review by igorkorsukov — APPROVED** [review]

**Pull request merged**